EventFilter/RPCRawToDigi/plugins/RPCTwinMuxRawToDigi.cc

```diff
@@ -0,0 +1,563 @@
+                                            << " from " << _tm_link;
+            continue;
+        }
+
+        if (_link_record.getConnector() > (unsigned int)RPCLBLink::max_connector_) {
+            if (fill_counters_ && _bx_offset == 0) {
+                _counters.add(RPCAMCLinkCounters::input_connector_invalid_, _tm_link);
+            }
+            LogDebug("RPCTwinMuxRawToDigi") << "Skipping invalid Connector " << _link_record.getConnector()
+                                            << " for record " << _link << " (" << std::hex << _link_record.getRecord()
+                                            << " in " << _record.getRecord()[0] << ':' << _record.getRecord()[1] << std::dec
+                                            << ") from " << _tm_link;
+            continue;
+        }
+
+        _lb_link.setLinkBoard(_link_record.getLinkBoard());
+        _lb_link.setConnector(_link_record.getConnector());
+
+        RPCLBLinkMap::map_type::const_iterator _lb_link_it = es_lb_link_map_->getMap().find(_lb_link);
+        if (_lb_link_it == es_lb_link_map_->getMap().end()) {
+            if (fill_counters_ && _bx_offset == 0) {
+                _counters.add(RPCAMCLinkCounters::input_connector_not_used_, _tm_link);
+            }
+            LogDebug("RPCTwinMuxRawToDigi") << "Could not find " << _lb_link
+                                            << " for record " << _link << " (" << std::hex << _link_record.getRecord()
+                                            << " in " << _record.getRecord()[0] << ':' << _record.getRecord()[1] << std::dec
+                                            << ") from " << _tm_link;
+            continue;
+        }
+
+        if (_bx < _bx_min || _bx > _bx_max) {
+            continue;
+        }
+
+        if (fill_counters_ && _bx == 0) {
+            _counters.add(RPCAMCLinkCounters::amc_data_, RPCAMCLink(_fed, _amc_number));
+            _counters.add(RPCAMCLinkCounters::input_data_, _tm_link);
+        }
+
+        RPCFebConnector const & _feb_connector(_lb_link_it->second);
+        RPCDetId _det_id(_feb_connector.getRPCDetId());
+        unsigned int _channel_offset(_link_record.getPartition() ? 9 : 1); // 1-16
+        ::uint8_t _data(_link_record.getPartitionData());
+
+        for (unsigned int _channel = 0 ; _channel < 8 ; ++_channel) {
+            if (_data & (0x1 << _channel)) {
+                unsigned int _strip(_feb_connector.getStrip(_channel + _channel_offset));
+                if (_strip) {
+                    _digis.insert(std::pair<RPCDetId, RPCDigi>(_det_id, RPCDigi(_strip, _bx)));
+                    LogDebug("RPCTwinMuxRawToDigi") << "RPCDigi " << _det_id.rawId()
+                                                    << ", " << _strip << ", " << _bx;
+                }
+            }
+        }
+
+        /*
+        if (fill_counters_ && _bx_offset == 0) {
+            rpctwinmux::RPCBXRecord _bx_record(_record.getRPCBXRecord(_link));
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant